lib/compress.js

```diff
@@ -479,7 +479,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant()) {
+                    if (!(stat instanceof AST_ForIn) && var_decl.value.is_constant()) {
                         var ctt = new TreeTransformer(function(node) {
                             if (node === ref) {
                                 var parent = ctt.parent();
```

Comment:
Do you mean to remove/revert this part like you do in https://github.com/mishoo/UglifyJS2/issues/1537#issuecomment-283831926?